libbeat/processors/add_cloud_metadata/provider_azure_vm.go

```diff
@@ -51,10 +85,147 @@ var azureVMMetadataFetcher = provider{
 				},
 				"region": c.Str("location"),
 			}.Apply(m)
-			return mapstr.M{"cloud": out}
+
+			azure, _ := s.Schema{
+				"resourcegroup": s.Object{
+					"name": c.Str("resourceGroupName"),
+				},
+			}.Apply(m)
+			return mapstr.M{"cloud": cloud, "azure": azure}
 		}
 
-		fetcher, err := newMetadataFetcher(config, "azure", azHeaders, metadataHost, azSchema, azMetadataURI)
-		return fetcher, err
+		azGenSchema := func(m map[string]interface{}) mapstr.M {
+			orchestrator := mapstr.M{
+				"orchestrator": mapstr.M{},
+			}
+
+			orchestrator.DeepUpdate(m)
+			return orchestrator
+		}
+
+		// hfetcher represents an http fetcher to retrieve metadata from azure metadata endpoint
+		hfetcher, err := newMetadataFetcher(config, "azure", azHeaders, metadataHost, azHttpSchema, azMetadataURI)
+		if err != nil {
+			return hfetcher, fmt.Errorf("failed to create new http metadata fetcher: %w", err)
+		}
+		// fetcher represents an azure metadata fetcher. The struct includes two type of fetchers.
+		// 1. An http fetcher(hfetcher) which retrieves metadata from azure metadata endpoint and
+		// 2. A generic fetcher(gfetcher) which uses azure sdk to retrieve metadata of azure managed clusters.
+		fetcher, err := newAzureMetadataFetcher("azure", hfetcher)
+		if err != nil {
+			return fetcher, fmt.Errorf("failed to create new azure metadata fetcher: %w", err)
+		}
+		// gfetcher is created and assinged to fetcher after the fetcher is created in order the
+		// fetchAzureClusterMeta to be a method of fetcher. This is needed so that the generic fetcher
+		// can use the results/metadata that are already retrieved from http fetcher. SubscriptionId and
+		// resourceGroupName are then used to filter azure managed clusters results.
+		gfetcher, err := newGenericMetadataFetcher(config, "azure", azGenSchema, fetcher.fetchAzureClusterMeta)
+		if err != nil {
+			return fetcher, fmt.Errorf("failed to create new generic metadata fetcher: %w", err)
+		}
+		fetcher.genericMetadataFetcher = gfetcher
+		return fetcher, nil
 	},
 }
+
+// fetchMetadata fetches azure vm metadata from
+// 1. Azure metadata endpoint with httpMetadataFetcher
+// 2. Azure Managed Clusters using azure sdk  with genericMetadataFetcher
```

Comment:
To make this render nicely in godoc, indent these two items.

```suggestion
//   1. Azure metadata endpoint with httpMetadataFetcher
//   2. Azure Managed Clusters using azure sdk  with genericMetadataFetcher
```